Lib/tarfile.py

```diff
@@ -2509,28 +2539,53 @@ def main():
     elif args.create is not None:
         tar_name = args.create.pop(0)
         _, ext = os.path.splitext(tar_name)
-        compressions = {
-            # gz
-            '.gz': 'gz',
-            '.tgz': 'gz',
-            # xz
-            '.xz': 'xz',
-            '.txz': 'xz',
-            # bz2
-            '.bz2': 'bz2',
-            '.tbz': 'bz2',
-            '.tbz2': 'bz2',
-            '.tb2': 'bz2',
-        }
         tar_mode = 'w:' + compressions[ext] if ext in compressions else 'w'
         tar_files = args.create
+        if args.verbose:
+            print('Create {!r}:'.format(tar_name))
+        # Actual adding files is done later.
+    elif args.add is not None:
+        tar_name = args.add.pop(0)
+        _, ext = os.path.splitext(tar_name)
 
-        with TarFile.open(tar_name, tar_mode) as tf:
-            for file_name in tar_files:
-                tf.add(file_name)
-
+        tar_mode = 'a'
+        tar_files = args.add
         if args.verbose:
-            print('{!r} file created.'.format(tar_name))
+            print('Add to {!r}:'.format(tar_name))
+
+        if ext in compressions:
+            parser.exit(
+                1,
+                "Error: Add is not a supported operation for compressed tar files")
```

Comment:
There currently isn't an option to unpack to tar file and in many cases unpacking would probably defeat the point of an append. I could possibly add an uncompress option to extract the tar file from the compressed formats then this might be an option.
Removed error if file does not exist, as it is then a create, and added the need to unpack and create.